build.py

```diff
@@ -143,29 +143,44 @@ def _establish_who_we_are(self):
 
     def doIt(self):
         self._establish_who_we_are()
+        self.validate()
+        self.prepare()
         if args.release == True:
             self.release()
         else:
             self.build()
+        self.tidy_up()
         self.report()
 
-    def build(self):
+    def validate(self):
+        if args.release:
+            if self.vars.version is None:
+                raise ValueError('Must set --version when releasing')
+            if self.vars.post_version is None:
+                raise ValueError('Must set --post-version when releasing')
+
+    def prepare(self):
+        if self.vars.version is not None:
+            self.set_version()
         self.determine_version()
+
+    def tidy_up(self):
+        if self.vars.post_version is not None:
+            self.set_post_version()
+            if args.release:
+                self.push_post_release_git_changes()
+
+    def build(self):
         self.run_unit_tests()
         self.build_python_wheel()
         self.pkg_docs()
 
     def release(self):
-        if self.vars.version is None:
-            raise ValueError('Must set --version when releasing')
         if self.vars.post_version is None:
```

Comment:
I guess we don't need these 2 lines either now as this is covered in `validate`? Looks like I missed this off my change in `lmctl`.

Reply:
I just assumed they was left there for another reason. I've removed them now 👍

Reply:
No was just my mistake. Ok great, will approve 👍